flang/runtime/environment.cpp

```diff
@@ -143,6 +143,17 @@ void ExecutionEnvironment::Configure(int ac, const char *av[],
     }
   }
 
+  if (auto *x{std::getenv("CUDA_STACKLIMIT")}) {
+    char *end;
+    auto n{std::strtol(x, &end, 10)};
+    if (n >= 0 && n < std::numeric_limits<int>::max() && *end == '\0') {
```

Comment:
cudaStackLimit is declared with `std::size_t` below; testing against the max signed int may be too restrictive.

Reply:
Right! I copy pasted that from above without really think about it. I'll update.